tests/src/python/test_provider_ogr_gpkg.py

```diff
@@ -2472,6 +2472,106 @@ def testIsSqlQuery(self):
         # Test flags
         self.assertFalse(vl2.vectorLayerTypeFlags() & Qgis.VectorLayerTypeFlag.SqlQuery)
 
+    def testCircularStringOnlyInUnknownTypeLayer(self):
+        """ Test bugfix for https://github.com/qgis/QGIS/issues/47610 """
+
+        tmpfile = os.path.join(self.basetestpath, 'testCircularStringOnlyInUnknownTypeLayer.gpkg')
+        ds = ogr.GetDriverByName('GPKG').CreateDataSource(tmpfile)
+        lyr = ds.CreateLayer('test', geom_type=ogr.wkbUnknown)
+        f = ogr.Feature(lyr.GetLayerDefn())
+        f.SetGeometry(ogr.CreateGeometryFromWkt('CIRCULARSTRING(0 0,1 1,2 0)'))
+        lyr.CreateFeature(f)
+        ds = None
+
+        vl = QgsVectorLayer('{}'.format(tmpfile), 'test', 'ogr')
+        self.assertEqual(1, vl.dataProvider().subLayerCount())
+        self.assertEqual(vl.dataProvider().subLayers(),
```

Comment:
subLayerCount/subLayers is the old approach to handling this -- can you also test QgsProviderRegistry.instance().providerMetadata('ogr').querySublayers(...) too?

Reply:
> QgsProviderRegistry.instance().providerMetadata('ogr').querySublayers(...) too?

after having tried, that's not super relevant because querySublayers() doesn't try to resolve wkbUnknown geometry type to the actualy geometry types taken by the features